### ISSUE TITLE
[MacOS] DateTime.ToUniversalTime performance issue

### ISSUE THREAD

**Issue by MarinAtanasov** [dotnet/coreclr#11652]
Hello,

I noticed recently that there is a huge difference in the performance of DateTime.ToUniversalTime between Windows and MacOS environments.

I have a test which calls 50000 times this method on a UTC and Local DateTime objects. On Windows, it runs for about 14ms, on MacOS -  240ms. The Windows PC might be about 2 times faster, but not 17 times.

Could you check if you can reproduce this? I am using netcoreapp1.1

Thanks,
Marin

**Comment by danmosemsft** [comment]
@MarinAtanasov are you able to try 2.0 preview also? how does that compare? There were some significant improvements relating to timezones that may be relevant.

**Comment by MarinAtanasov** [comment]
@danmosemsft cannot test because 2.0 preview requires Sierra. I will be staying with El Capitan for at least a few more weeks.

**Comment by danmosemsft** [comment]
@tarekgh is it reasonable to close this as fixed?

**Comment by tarekgh** [comment]
@eerhardt was the one working on that so I think he can confirm.

**Comment by eerhardt** [comment]
I just ran the following code on my macOS Sierra machine and my win10 machine using:

```
Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview2-25319-02
  Build    : 7983b575ebcbdc3a825eea4b141ff7fb69d25f9d
```

```C#
            DateTime dt = DateTime.Now;
            DateTime utc = DateTime.UtcNow;
            for (int i = 0; i < 20; i++)
            {
                Stopwatch watch = new Stopwatch();
                watch.Start();

                for (int j = 0; j < 50_000; j++)
                {
                    dt.ToUniversalTime();
                    utc.ToUniversalTime();
                }
                Console.Write(watch.ElapsedMilliseconds + "<br>");
            }
```

**MacBook Pro** | **Windows 10** 
----------------- | -------------------------------------
21<br>24<br>23<br>24<br>23<br>24<br>24<br>24<br>23<br>23<br>23<br>23<br>23<br>24<br>24<br>24<br>27<br>24<br>24<br>24<br> | 16<br>17<br>14<br>15<br>13<br>17<br>15<br>16<br>15<br>15<br>14<br>17<br>15<br>17<br>16<br>16<br>19<br>16<br>15<br>17<br>

Closing.

**Comment by danmosemsft** [comment]
Excellent thank you @eerhardt .

**Comment by MarinAtanasov** [comment]
Thanks for the quick response guys! Glad to hear that it is OK in 2.0.

**Comment by karelz** [comment]
In general: MacOS is development scenario, so I don't think that having something slower (even 17x) is a big deal (although fixed in this case in 2.0), unless it impacts speed of significant end-to-end scenario (e.g. ability to build, debug, deploy, or run basic tests).

**Comment by eerhardt** [comment]
Just an FYI - TimeZoneInfo has the same implementation on macOS and Linux, so the perf bugs that I fixed were affecting both platforms.
